File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/core/usecase/usecases/base/BaseMyShopUseCase.java
package com.mycompany.myshop.testkit.dsl.core.usecase.usecases.base;

import com.mycompany.myshop.testkit.driver.port.MyShopDriver;
import com.mycompany.myshop.testkit.dsl.core.shared.BaseUseCase;
import com.mycompany.myshop.testkit.dsl.core.shared.UseCaseContext;

public abstract class BaseMyShopUseCase<R, V> extends BaseUseCase<MyShopDriver, R, V> {
    protected BaseMyShopUseCase(MyShopDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}

File: system-test/typescript/src/testkit/dsl/core/usecase/usecases/base/BaseMyShopUseCase.ts
import type { MyShopDriver } from '../../../../../driver/port/my-shop-driver.js';
import { BaseUseCase } from '../../../shared/base-use-case.js';

export abstract class BaseMyShopUseCase<TResponse, TVerification> extends BaseUseCase<
  MyShopDriver,
  TResponse,
  TVerification
> {}
